moto/cloudhsmv2/models.py

```diff
@@ -0,0 +1,312 @@
+
+        for new_tag in tag_list:
+            tag_exists = False
+            for existing_tag in self.tags[resource_id]:
+                if existing_tag["Key"] == new_tag["Key"]:
+                    existing_tag["Value"] = new_tag["Value"]
+                    tag_exists = True
+                    break
+            if not tag_exists:
+                self.tags[resource_id].append(new_tag)
+
+        return {}
+
+    def untag_resource(
+        self, resource_id: str, tag_key_list: List[str]
+    ) -> Dict[str, Any]:
+        if resource_id in self.tags:
+            self.tags[resource_id] = [
+                tag for tag in self.tags[resource_id] if tag["Key"] not in tag_key_list
+            ]
+
+        return {}
+
+    def create_cluster(
+        self,
+        backup_retention_policy: Optional[Dict[str, str]],
+        hsm_type: str,
+        source_backup_id: Optional[str],
+        subnet_ids: List[str],
+        network_type: Optional[str],
+        tag_list: Optional[List[Dict[str, str]]],
+        mode: Optional[str],
+    ) -> Dict[str, Any]:
+        cluster = Cluster(
+            backup_retention_policy=backup_retention_policy,
+            hsm_type=hsm_type,
+            source_backup_id=source_backup_id,
+            subnet_ids=subnet_ids,
+            network_type=network_type or "IPV4",
+            tag_list=tag_list,
+            mode=mode or "DEFAULT",
+            region_name=self.region_name,
+        )
+        self.clusters[cluster.cluster_id] = cluster
+
+        backup = Backup(
+            cluster_id=cluster.cluster_id,
+            hsm_type=hsm_type,
+            mode=mode or "DEFAULT",
+            tag_list=tag_list,
+            region_name=self.region_name,
+        )
+        self.backups[backup.backup_id] = backup
+
+        return cluster.to_dict()
+
+    def delete_cluster(self, cluster_id: str) -> Dict[str, Any]:
+        if cluster_id not in self.clusters:
+            raise ResourceNotFoundException(f"Cluster {cluster_id} not found")
+
+        cluster = self.clusters[cluster_id]
+        cluster.state = "DELETED"
+        cluster.state_message = "Cluster deleted"
+        del self.clusters[cluster_id]
+        return cluster.to_dict()
+
+    def describe_clusters(
+        self, filters: Dict[str, List[str]], next_token: str, max_results: int
+    ) -> Tuple[List[Dict[str, Any]], Optional[str]]:
+        clusters = list(self.clusters.values())
+
+        if filters:
+            for key, values in filters.items():
+                if key == "clusterIds":
+                    clusters = [c for c in clusters if c.cluster_id in values]
+                elif key == "states":
+                    clusters = [c for c in clusters if c.state in values]
+                elif key == "vpcIds":
+                    clusters = [c for c in clusters if c.vpc_id in values]
+
+        clusters = sorted(clusters, key=lambda x: x.create_timestamp)
+
+        if not max_results:
```

Comment:
This doesn't look right - even if we don't have `max_results` specified, we may have a `next_token` that we should honor.

Can we use the `@paginate`-decorator here instead?